### PR TITLE
Unpin alembic

### PR DESCRIPTION
**Pull request by harupy**
## What changes are proposed in this pull request?

Unpin alembic and fix migration scripts affected by this change.

Closes #5245, #4810, #4215

## How is this patch tested?

Existing tests

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/store/db_migrations/versions/0a8213491aaa_drop_duplicate_killed_constraint.py [hunk 1]
```diff
@@ -31,7 +31,8 @@ def upgrade():
     # operation is expected to fail under certain circumstances, we execute `drop_constraint()`
     # outside of the batch operation context.
     try:
-        op.drop_constraint(constraint_name="status", table_name="runs", type_="check")
+        # op.drop_constraint(constraint_name="status", table_name="runs", type_="check")
```
Comment (WeichenXu123) [inline mlflow/store/db_migrations/versions/0a8213491aaa_drop_duplicate_killed_constraint.py:34]: Why drop this ?

Comment (harupy) [reply mlflow/store/db_migrations/versions/0a8213491aaa_drop_duplicate_killed_constraint.py:34]: It's just temporarily commented out.

Comment (harupy) [reply mlflow/store/db_migrations/versions/0a8213491aaa_drop_duplicate_killed_constraint.py:34]: sqlite:

```sql
CREATE TABLE runs (
	run_uuid VARCHAR(32) NOT NULL,
	name VARCHAR(250),
	source_type VARCHAR(20),
	source_name VARCHAR(500),
	entry_point_name VARCHAR(50),
	user_id VARCHAR(256),
	status VARCHAR(9),
	start_time BIGINT,
	end_time BIGINT,
	source_version VARCHAR(50),
	lifecycle_stage VARCHAR(20),
	artifact_uri VARCHAR(200),
	experiment_id INTEGER,
	CONSTRAINT run_pk PRIMARY KEY (run_uuid),
	FOREIGN KEY(experiment_id) REFERENCES experiments (experiment_id),
	CONSTRAINT runs_lifecycle_stage CHECK (lifecycle_stage IN ('active', 'deleted')),
	CONSTRAINT source_type CHECK (source_type IN ('NOTEBOOK', 'JOB', 'LOCAL', 'UNKNOWN', 'PROJECT')),
	CHECK (status IN ('SCHEDULED', 'FAILED', 'FINISHED', 'RUNNING', 'KILLED'))
)
```

postgres:

```sql
CREATE TABLE runs (
	run_uuid VARCHAR(32) NOT NULL,
	name VARCHAR(250),
	source_type VARCHAR(20),
	source_name VARCHAR(500),
	entry_point_name VARCHAR(50),
	user_id VARCHAR(256),
	status VARCHAR(9),
	start_time BIGINT,
	end_time BIGINT,
	source_version VARCHAR(50),
	lifecycle_stage VARCHAR(20),
	artifact_uri VARCHAR(200),
	experiment_id INTEGER,
	CONSTRAINT run_pk PRIMARY KEY (run_uuid),
	CONSTRAINT runs_experiment_id_fkey FOREIGN KEY(experiment_id) REFERENCES experiments (experiment_id),
	CONSTRAINT source_type CHECK ((source_type)::text = ANY ((ARRAY['NOTEBOOK'::character varying, 'JOB'::character varying, 'LOCAL'::character varying, 'UNKNOWN'::character varying, 'PROJECT'::character varying])::text[])),
	CONSTRAINT runs_lifecycle_stage CHECK ((lifecycle_stage)::text = ANY ((ARRAY['active'::character varying, 'deleted'::character varying])::text[])),
	CONSTRAINT runs_status_check CHECK ((status)::text = ANY ((ARRAY['SCHEDULED'::character varying, 'FAILED'::character varying, 'FINISHED'::character varying, 'RUNNING'::character varying, 'KILLED'::character varying])::text[]))
)
```

Comment (harupy) [reply mlflow/store/db_migrations/versions/0a8213491aaa_drop_duplicate_killed_constraint.py:34]: - Both sqlite and postgres have a check constraint for status.

Comment (harupy) [reply mlflow/store/db_migrations/versions/0a8213491aaa_drop_duplicate_killed_constraint.py:34]: Looks like we need this operation for mysql.

**Review by dbczumar — COMMENTED** [review]
@harupy Looks good, just need to address the check constraint issue for MySQL. Thanks for doing this!

**Comment by harupy** [comment]
Confirmed the new migration script works for both old and new versions of alembic.

**Comment by harupy** [comment]
@WeichenXu123 @dbczumar Could you take a look again and approve?

File: .github/workflows/master.yml [hunk 2]
```diff
@@ -161,21 +161,20 @@ jobs:
...
-        docker-compose run mlflow-postgres python run_checks.py --schema-output schemas/postgres.sql
-        docker-compose run mlflow-mysql python run_checks.py --schema-output schemas/mysql.sql
-        docker-compose run mlflow-mssql ./init-mssql-db.sh
-        docker-compose run mlflow-mssql python run_checks.py --schema-output schemas/mssql.sql
-        docker-compose down --rmi all --volumes --remove-orphans
+        docker-compose run mlflow-sqlite
+        docker-compose run mlflow-postgres
+        docker-compose run mlflow-mysql
+        docker-compose run mlflow-mssql
+        docker-compose down --volumes --remove-orphans --rmi all
```
Comment (harupy) [inline .github/workflows/master.yml:177]: Cleaned up this step by adding commands in docker-compose.yml.

File: tests/db/Dockerfile [hunk 3]
```diff
@@ -2,8 +2,7 @@ FROM python:3.6
 
 WORKDIR /tmp/mlflow
 
-COPY dist ./dist
-
-RUN pip install dist/*.whl
 RUN pip install psycopg2 pymysql mysqlclient
+COPY dist ./dist
+RUN pip install dist/mlflow-*.whl
```
Comment (harupy) [inline tests/db/Dockerfile:7]: For rebuilding the image a bit faster.

File: tests/db/docker-compose.yml [hunk 4]
```diff
@@ -17,15 +17,17 @@ services:
...
   mysql:
-    image: mysql:5.7
+    image: mysql
     restart: always
     environment:
       MYSQL_ROOT_PASSWORD: root-password
       MYSQL_DATABASE: mlflowdb
       MYSQL_USER: mlflowuser
       MYSQL_PASSWORD: mlflowpassword
+    command: mysqld --default-authentication-plugin=mysql_native_password
```
Comment (harupy) [inline tests/db/docker-compose.yml:30]: In MySQL >= 8.0.4, this command is required to log in using a password.

File: tests/db/run_checks.py [hunk 5]
```diff
@@ -35,6 +35,12 @@ def run_logging_operations():
         )
         print(mlflow.get_run(run.info.run_id))
 
+    # Ensure the following migration scripts work correctly:
+    # - cfd24bdc0731_update_run_status_constraint_with_killed.py
+    # - 0a8213491aaa_drop_duplicate_killed_constraint.py
+    client = mlflow.tracking.MlflowClient()
+    client.set_terminated(run_id=run.info.run_id, status="KILLED")
```
Comment (harupy) [inline tests/db/run_checks.py:42]: Added a check to ensure the updated migration scripts work properly.

File: .github/workflows/master.yml [hunk 6]
```diff
@@ -161,21 +161,20 @@ jobs:
...
     # Separate build and run to make it easier to explore logs
     - name: Run database initialization tests - build
+      working-directory: tests/db
       run: |
-        python setup.py bdist_wheel
-        cp -r dist tests/db
-        cd tests/db
+        ./build_wheel.sh
         docker-compose pull
+        docker image ls | grep -E '(REPOSITORY|postgres|mysql|mssql)'
```
Comment (harupy) [inline .github/workflows/master.yml:168]: Show database image versions for debugging.

File: setup.py [hunk 7]
```diff
@@ -65,7 +65,7 @@ def package_files(directory):
 other capabilities.
 """
 CORE_REQUIREMENTS = SKINNY_REQUIREMENTS + [
-    "alembic<=1.4.1",
+    "alembic",
```
Comment (harupy) [inline setup.py:68]: 🎉

**Review by dbczumar — APPROVED** [review]
LGTM!

**Pull request merged**